Olive.Microservices.Hub/Controllers/Modules/Components/ManualLogin.cs

```diff
@@ -0,0 +1,131 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.ComponentModel.DataAnnotations;
+using System.Linq;
+using System.Threading.Tasks;
+using System.Transactions;
+using System.Web;
+using Olive;
+using Olive.Entities;
+using Olive.Mvc;
+using Olive.Web;
+using Domain;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Filters;
+using Microsoft.AspNetCore.Mvc.ModelBinding;
+using Microsoft.AspNetCore.Mvc.ModelBinding.Validation;
+using Microsoft.AspNetCore.Mvc.Rendering;
+using Microsoft.AspNetCore.Mvc.ViewFeatures;
+using PeopleService;
+using vm = ViewModel;
+
+namespace ViewComponents
+{
+    [EscapeGCop("Auto generated code.")]
+#pragma warning disable
+    public partial class ManualLogin : ViewComponent
+    {
+        public async Task<IViewComponentResult> InvokeAsync(vm.ManualLogin info)
+        {
+            return View(await Bind<vm.ManualLogin>(info));
+        }
+    }
+}
+
+namespace Controllers
+{
+    [EscapeGCop("Auto generated code.")]
+#pragma warning disable
+    public partial class ManualLoginController : BaseController
+    {
+        [HttpPost("ManualLogin/SimulateLogIn")]
+        [ValidateAntiForgeryToken]
+        public async Task<ActionResult> SimulateLogIn(vm.ManualLogin info)
+        {
+            if (!(info.IsVisible))
+                return new UnauthorizedResult();
+            await info.CopyDataTo(info.Item);
+
+            info.Item.Roles = info.RoleNames;
+
+            await info.Item.LogOn();
+
+            return Redirect(Request.Param("returnUrl"));
+        }
+
+        [NonAction, OnPreBinding]
+        public async Task OnPreBinding(vm.ManualLogin info)
+        {
+            if (Request.IsGet())
+            {
+                // Set default roles
+                info.RoleNames = @"Employee,Dev,JuniorDev,SeniorDev,LeadDev,HeadDev,QA,JuniorQA,SeniorQA,LeadQA,HeadQA,BA,JuniorBA,SeniorBA,LeadBA,HeadBA,PM,JuniorPM,SeniorPM,LeadPM,HeadPM,AM,JuniorAM,SeniorAM,LeadAM,HeadAM,Director,JuniorDirector,SeniorDirector,LeadDirector,HeadDirector,Designer,JuniorDesigner,SeniorDesigner,LeadDesigner,HeadDesigner,IT,JuniorIT,SeniorIT,LeadIT,HeadIT,Reception,JuniorReception,SeniorReception,LeadReception,HeadReception,PA,JuniorPA,SeniorPA,LeadPA,HeadPA,Sales,JuniorSales,SeniorSales,LeadSales,HeadSales";
+            }
+        }
+
+        [NonAction, OnBound]
+        public async Task OnBound(vm.ManualLogin info)
+        {
+            info.Item = info.Item ?? new UserInfo();
+
+            info.IsVisible = info.AllowManual;
+
+            if (Request.IsGet()) await info.Item.CopyDataTo(info);
+
+            info.DisplayName = "Paymon Khamooshi";
+            info.Email = "paymon@geeks.ltd.uk";
+            info.Item.ID = "69D81564-809D-45A8-83CC-D12971A008D5".To<Guid>();
```

Comment:
@PaymonK I have hardcoded your ID as in some micro-services we need a valid person who has a valid ID in the People micro-service. This ID is coming from the People microservice.